### PR TITLE
Keyboard up/down key issue in Channel switcher due to global thread list

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Moving up/down doesn’t work as expected in find channels (ctrl/cmd + k) if global threads is in background, adding more specific conditions in threads list keyboard event listener to avoid this.

#### Ticket Link

  Fixes https://mattermost.atlassian.net/browse/MM-40444

#### Related Pull Requests
``NONE``

#### Screenshots
|  before  |  after  |
|----|----|
| https://user-images.githubusercontent.com/16203333/155275448-2d681acc-2eca-4822-95d9-a0c163fee270.mp4 | https://user-images.githubusercontent.com/16203333/155275473-407afcec-7249-4d47-b2f9-b1066938a271.mp4 |

#### Release Note
```release-note
*UI: up/down key in find channel when global threads are in background
```

### CODE REVIEW

File: e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                });
+
+            // * the button on the RHS header should now say Following
+            cy.get('#rhsContainer').find('.FollowButton').should('have.text', 'Following');
+
+            // # Visit global threads
+            cy.uiClickSidebarItem('threads');
+
+            // * There should be a thread there
+            cy.get('article.ThreadItem').should('have.have.lengthOf', 1);
```
Comment (koox00) [inline e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js:84]: I think all these are tested in another test case.
No need to add them here as well.
We could just create a couple of threads and then navigate to global threads.

File: e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        // * Confirm the offtopic channel is selected in the suggestion list
+        cy.get('#suggestionList').findByTestId('off-topic').should('be.visible').and('have.class', 'suggestion--selected');
+
+        // # Press up arrow
+        cy.wait(TIMEOUTS.HALF_SEC);
+        cy.get('body').type('{uparrow}');
+
+        // * Confirm the townsquare channel is selected in the suggestion list
+        cy.get('#suggestionList').findByTestId('town-square').should('be.visible').and('have.class', 'suggestion--selected');
```
Comment (koox00) [inline e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js:103]: Can you also assert that the URL doesn't change?
i.e: The URL would change if threads list consumed the "up/down arrow" event.
We'll need 2 threads for that.

File: e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        cy.get('body').type('{downarrow}');
+
+        // * Confirm the offtopic channel is selected in the suggestion list
+        cy.get('#suggestionList').findByTestId('off-topic').should('be.visible').and('have.class', 'suggestion--selected');
+
+        // # Press ENTER
+        cy.get('body').type('{enter}');
+
+        // * Confirm that channel is open, and post text box has focus
+        cy.contains('#channelHeaderTitle', 'Off-Topic');
```
Comment (koox00) [inline e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js:116]: You could also assert that URL is the expected

```js
         cy.url().should('include', '/channels/off-topic');
```

0/5

**Review by koox00 — APPROVED** [review]
LGTM!
I'm just a bit concerned, we have many events added on document, and clashes won't be avoided completely.

Just a thought. Would it be worth to consider event delegation, and emitting custom events depending on the case, just for document events?

**Review by hmhealey — APPROVED** [review]

**Comment by AshishDhama** [comment]
e2e-test

File: e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            });
+        });
+    });
+
+    beforeEach(() => {
+        // # Visit channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+    });
+
+    it('MM-T1243 CTRL/CMD+K - Create thread, Open global threads and then from find channels switch channel using arrow keys and Enter', () => {
```
Comment (jgilliam17) [inline e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js:48]: @AshishDhama Is this a new test case? If, yes I need to create a new test case ID

Comment (AshishDhama) [reply e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js]: @jgilliam17 Thanks for pointing this out I think have forgotten to add the new one. I have updated now.

Comment (AshishDhama) [reply e2e/cypress/integration/keyboard_shortcuts/ctrl_cmd_k_open_channel_from_global_threads_spec.js]: @jgilliam17 Thanks for pointing this out I think have forgotten to add the new one. I have updated it now.

**Comment by jgilliam17** [comment]
Thanks @AshishDhama 
Tested manually and this issue is fixed.
- Verified while on global threads and using Find Channels modal, tapping up/down keys to navigate through channel list is working as expected.
One issue - I am not able to find the cypress test case on the [lates run ](https://mattermost-cypress-report.s3.amazonaws.com/155910-835cbc0-onprem-ent-webapp-pr-9685-mattermost/mm-ee-test835cbc0/mochawesome.html) It does not appear that spec file ran at all. Can you please take a look - thanks

**Review by jgilliam17 — APPROVED** [review]
Thanks @AshishDhama 
Looks good to merge.
Verified new cypress test is passing
Screen Shot 2022-02-28 at 12 38 29 PM

**Pull request merged**